### DISCUSSION TITLE
A Query regarding the development of decentralised identity on LokiNet

### DISCUSSION THREAD

**Discussion by gokulgaze** [oxen-io/lokinet#1687] [Q&A]
Dear LokiNet team,

I am trying to integrate lokinet relayer network to secure and scale a web3 based DID system known as Ceramic to ensure the protection of the personal identifiable information. I have already setup lokinet relayers on multiple machines. How can I create a network of these relayers and configure a web application authorisation based on lokinet. Does lokinet offer identity based encryption (IBE) or Attribute based Encryption (ABE) for these kind of scenarios.

**Comment by majestrate** [comment]
In lokinet our primary "api" is dns. `.loki` addresses are base32 encoded ed25519 public keys. All locally mapped ip addresses have a PTR record that gives the `.loki` address they are mapped to, forward and reverse dns match. all traffic to and from `.loki` are both end to end encrypted and authenticated.

I do not know what your requirements are but lokinet may already provide most of what you are looking for, the rest would be making your application aware of `.loki` addresses using dns. I have a repo that demonstrates "lokinet awareness" [here](https://github.com/majestrate/lokinet-aware-demos) that could use a bit more explanation, but the code is there.
